src/introduction/hal-design-philosophy.md

```diff
@@ -1 +1,17 @@
 # HAL Design Philosophy
+
+The HAL was designed based on some recommended patterns and rules that are described in several documents:
+- [`esp-hal` developers' guidelines][guidelines]
+- [The Embedded Rust Book - HAL Design Patterns][embedded-rust-patterns]
+
+[guidelines]: https://github.com/esp-rs/esp-hal/blob/main/documentation/DEVELOPER-GUIDELINES.md
+[embedded-rust-patterns]: https://docs.rust-embedded.org/book/design-patterns/hal/index.html
+
+The HAL implements both blocking and asynchronous APIs for the supported peripherals. When applicable, drivers implement the [`embedded-hal`][embedded-hal] and [`embedded-hal-async`][embedded-hal-async] traits.
```

Comment:
```suggestion
The HAL implements blocking APIs for all peripherals, and asynchronous APIs for the peripherals which support asynchronous operation. When applicable, drivers implement the [`embedded-hal`][embedded-hal] and [`embedded-hal-async`][embedded-hal-async] traits.
```

My suggestion is a bit repetitive I guess, but this could use a bit of clarification IMO. "Supported peripherals" is not very clear to me, is this referring to the drivers or the hardware itself?